### PR TITLE
Add partial MVP for diffdrive control

### PR DESCRIPTION
**Pull request by njreichert**
This PR introduces the majority of the systems required for motor control:

- Low-level drivers (Vendored in from [PiPCA9685](https://github.com/barulicm/PiPCA9685))
- Basic ros2_control controller logic (Modified from [ros2_control_demos](https://github.com/ros-controls/ros2_control_demos))
- Joystick support

Remaining items to do:

- [ ] Clean up / refactor package layout
- [x] Create launch files combining ros2_control systems, camera systems, and teleop

As an aside this commit also introduces a "unified" dockerfile that builds multiple targets. Given that the Robot image should ideally be a strict subset of the dev image, this change should help with dependency management.

### CODE REVIEW

**Comment by njreichert** [comment]
diffbot can be run on dev hardware with `ros2 launch ros2_control_demo_example_2 diffbot.launch.py use_mock_hardware:=True`

File: build_robot_image.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+docker build \
+    -t umrt-ros \
```
Comment (ahmeda16) [inline build_robot_image.sh]: Change tag to umrt-ros-robot for clarity

File: src/example_2/bringup/launch/joy.launch.py
Comment (ahmeda16) [inline src/example_2/bringup/launch/joy.launch.py:1]: Could we move joylaunch into diffbotlaunch? Looks small enough where it doesn't require its own

**Comment by ahmeda16** [comment]
Merge main into this branch to avoid conflicts

File: src/example_2/bringup/launch/joy.launch.py
Comment (ahmeda16) [reply src/example_2/bringup/launch/joy.launch.py:1]: This'll change some launch stuff too in the main launch but I think its worth while

Comment (njreichert) [reply src/example_2/bringup/launch/joy.launch.py:1]: I'd rather keep these separate so we can keep it more modular.

**Pull request merged**